Polish notebooks

emmarogge opened the pull request:
Extract shared utilities/layout/styling to make notebooks more readable. Update content of overview sections in each notebook and ensure header links point to the new directory (first_hour_on_vwb).

emmarogge committed:
Polish  First Hour on Verily Workbench notebooks

- Update GitHub, Vertex AI header links to reflect change in directory name.
- Change '\''Open in Terra notebook instance'\'' to '\''Open in Verily Workbench cloud environment'\''

emmarogge committed:
Extract widget utilities into python module

Extract widget utility functions, styling and layout into new Python module. Import module in groups notebook and style buttons, widgets with color from Verily brand kit.

emmarogge committed:
Refactor overview section of resources notebook

Update content of resources notebook and extract shared dependencies.

emmarogge committed:
Polish data collections notebook

Extract shared utilities and update overview content. Ensure that if a user wants to populate just the data collection conversion widget, the provided input workspace ID is used instead of the output from the previous widget.

johntbates commented on first_hour_on_vwb/widget_utils.py:
```diff
@@ -0,0 +1,92 @@
+    width='75%'
+)
+
+def users_table(stdout):
```
nit: Looks like you're parsing an arbitrary JSON string, not  just those from stdout.  Consider renaming the argument?

emmarogge replied:
done!

johntbates commented on first_hour_on_vwb/widget_utils.py:
```diff
@@ -0,0 +1,92 @@
+    html += "</table>"
+    return html
+
+class TextInputWidget(object):
```
nit and suggestion: I use and love Python [dataclasses](https://realpython.com/python-data-classes/).  They're super concise - especially w.r.t. the __init__ method since it's not required at all!  Here and throughout, you might consider using them.

emmarogge replied:
thank you for the suggestion -- I have refactored to use @dataclass for the widget-component classes!

johntbates commented on first_hour_on_vwb/widget_utils.py:
```diff
@@ -0,0 +1,92 @@
+
+class TextInputWidget(object):
+    def __init__(self, ph, desc):
+        self.description = desc
```
Here and in the other classes below, you set up some internal state in member vars like `self.description` and `self.placeholder` that don't appear to be referenced by anything other than the `self.widget` constructors.  Are these member vars actually necessary?

emmarogge replied:
They are necessary now that I've switched to creating each widget in a @dataclass-annotated class, within the `__post_init__` method.

johntbates reviewed (approved):
Generally LGTM, but please consider my comments about dataclasses and unused member vars.

emmarogge committed:
Updates due to reviewer feedback

Updates include:
- Rename input parameter of users_table method
- Refactor widget utils classes to use @dataclass

emmarogge committed:
Remove unnecessary import
